app/src/main/res/layout/row_item_pending_message.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  Copyright (c) 2016-2017 Onegini B.V.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  -->
+
+<android.support.v7.widget.CardView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_marginTop="@dimen/activity_margin"
+    android:layout_marginLeft="@dimen/activity_margin"
+    android:layout_marginRight="@dimen/activity_margin"
+    android:layout_height="wrap_content">
+
+    <RelativeLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:padding="8dp">
+
+        <TextView
+            android:id="@+id/pending_message_profile"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tools:text="Robert"
+            android:layout_alignParentTop="true"
+            android:layout_alignParentLeft="true"
+            android:layout_alignParentStart="true" />
+
+        <TextView
+            android:id="@+id/pending_message_timestamp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="12sp"
+            tools:text="14:27:36"
+            android:layout_alignParentTop="true"
+            android:layout_alignParentRight="true"
+            android:layout_alignParentEnd="true"/>
+
+        <TextView
+            android:id="@+id/pending_message"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="18sp"
+            android:textColor="@color/blue"
+            tools:text="Transaction message"
+            android:layout_below="@+id/pending_message_profile"
+            android:layout_alignParentLeft="true"
+            android:layout_alignParentStart="true"
+            android:layout_marginTop="4dp"
+            android:layout_marginBottom="4dp"/>
+
+        <TextView
+            android:id="@+id/pending_message_expires"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tools:text="Will expire at: 14:35:36"
```

Comment:
hardcoded text

Reply:
Once again, I treat it as a developer hint, not the app resource, just like a code comments. Because of that the hardcoded value is fine for me. Don't you agree?

Reply:
I will lease the "Will expire at: " part, but I would change the time to < expiration time > or sth... to show the developer it's parameterized

Reply:
IMO it should show an example of expected value, because you want to see the layout, how it looks like. In this case the string length is constant (the time always use HH:MM:SS pattern) so you can easly see how it fits in the layout.

Reply:
So in general I use `tools` to generate something that looks like a design provided from a designer. I know that we don't have a design for this box, but you know what I mean :)